upcoming: [M3-7296] - Create Load Balancer Summary page

cpathipa opened the pull request:
## Description 📝
Add's Load Balancer Summary page.

> [!note]
>  This PR does not support rules and pricing. I will cover that in a separate PR to keep the size down.

## Changes  🔄

![image](https://github.com/linode/manager/assets/119517080/0c2b1d66-c95a-4ac5-9e27-4b3c7ac14585)


## How to test 🧪

### Prerequisites
(How to setup test environment)
- Turn on mocks or test with the AGLB test account.
- Turn on the AGLB Full Create Flow feature flag.
- Go to http://localhost:3000/loadbalancers/create


### Verification steps 
(How to verify changes)
- Verify you can add, update, and remove LoadBalancer Label, Service Targets and Routes on the summary page of  AGLB create flow. 


## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

mjac0bs commented on packages/manager/.changeset/pr-10018-upcoming-features-1703185915854.md:
```diff
@@ -0,0 +1,5 @@
+"@linode/manager": Upcoming Features
+---
+
+Create Load Balancer Summary page. ([#10018](https://github.com/linode/manager/pull/10018))
```
```suggestion
Create Load Balancer Summary page ([#10018](https://github.com/linode/manager/pull/10018))
```

mjac0bs reviewed (commented):
✅ I could view the Load Balancer summary page and edit my Load Balancer label and configuration (label, protocol, port). 

❓ I know @bnussman-akamai mentioned a future meeting with UX to discuss some Create Flow issues, so maybe these are already on your/his radar, but:

- There isn't a way for a user to remove service targets and routes or delete an additional configuration in the create flow. If a user selects Add a Configuration and then decides they don't want that additional configuration, they're stuck with proceeding with it and filling out at least the label (required) to move to Review Load Balancer and can't delete it there either. Are there plans to address this in the stepper flow and/or Summary page?

Minor UX feedback:
- Currently the Edit Regions drawer has an enabled Save Changes button, even though these regions can't be edited for Beta. Curious if we wanted to disable that button for the time being or feel it's just not worth it. 
![Screenshot 2024-01-04 at 2 34 05 PM](https://github.com/linode/manager/assets/114685994/6593b85f-f7ce-42ab-819a-329b86fda96b)

- Any thoughts on the background for the Load Balancer Summary (white paper background) diverging from the design with the Cluster Summary (no background)? Wondering if it looks a little odd to have that much white space in tablet view. Or maybe we want that. But do we want Kube and Load Balancer to be different? Probably not. It would be nice to make UX aware of the discrepancy and get some feedback. 

https://github.com/linode/manager/assets/114685994/303de449-5f7b-49ff-bbc3-13dee8616457

bnussman-akamai reviewed (commented):
Clicking `Edit` on the Configuration causes the Accordion to open/close. 



https://github.com/linode/manager/assets/115251059/181275f2-f7b3-4358-922e-819bbe42e0f4

Can we put the headers and the `Edit` buttons on the same line?
![Screenshot 2024-01-12 at 10 46 05 AM](https://github.com/linode/manager/assets/115251059/4204df08-384d-456a-9ad5-340b28b300f6)

cpathipa commented:
@bnussman-akamai @mjac0bs Thank you for the feedback. I will review and address the issues

cpathipa commented on packages/manager/src/features/LoadBalancers/LoadBalancerCreate/LoadBalancerSummary/LoadBalancerSummary.tsx:
```diff
@@ -0,0 +1,91 @@
+            <Notice important variant="warning">
+              <Typography>
+                Upload and apply certificates after LB creation.
+                <Link to="#">Learn more</Link>.
```
Addressed in [1a8baae](https://github.com/linode/manager/pull/10018/commits/1a8baae7cf85149874a5b8c7b45fa40fd65f80ee)

cpathipa commented on packages/manager/src/features/LoadBalancers/LoadBalancerCreate/LoadBalancerSummary/LoadBalancerSummary.tsx:
```diff
@@ -0,0 +1,91 @@
+            <Divider dark />
+            <Notice important variant="warning">
+              <Typography>
+                Upload and apply certificates after LB creation.
```
I don't think this is the final copy. "Load Balancer" makes more sense instead of "LB".
addressed spacing 5cb1279
![image](https://github.com/linode/manager/assets/119517080/448f541a-24a7-4775-8e40-2be0423f3aa8)

cpathipa commented:
> ✅ I could view the Load Balancer summary page and edit my Load Balancer label and configuration (label, protocol, port).
> 
> ❓ I know @bnussman-akamai mentioned a future meeting with UX to discuss some Create Flow issues, so maybe these are already on your/his radar, but:
> 
> * There isn't a way for a user to remove service targets and routes or delete an additional configuration in the create flow. If a user selects Add a Configuration and then decides they don't want that additional configuration, they're stuck with proceeding with it and filling out at least the label (required) to move to Review Load Balancer and can't delete it there either. Are there plans to address this in the stepper flow and/or Summary page?
> 
> Minor UX feedback:
> 
> * Currently the Edit Regions drawer has an enabled Save Changes button, even though these regions can't be edited for Beta. Curious if we wanted to disable that button for the time being or feel it's just not worth it.
>   ![Screenshot 2024-01-04 at 2 34 05 PM](https://private-user-images.githubusercontent.com/114685994/294314598-6593b85f-f7ce-42ab-819a-329b86fda96b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwODAxODksIm5iZiI6MTcwNTA3OTg4OSwicGF0aCI6Ii8xMTQ2ODU5OTQvMjk0MzE0NTk4LTY1OTNiODVmLWY3Y2UtNDJhYi04MTlhLTMyOWI4NmZkYTk2Yi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMlQxNzE4MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02Y2ZhMWMzMzU2ZWE4YzMzZGY0MDNkOTg0ZmUwOWFmNzg2ZTg1MjAyM2E1Zjk1OGI2ZjE1MzliZDllZjhiZjdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.T-f1uwXjsbUm1Kbx6cmyjDEzMjYQuny-7w_0vG0XYws)
> * Any thoughts on the background for the Load Balancer Summary (white paper background) diverging from the design with the Cluster Summary (no background)? Wondering if it looks a little odd to have that much white space in tablet view. Or maybe we want that. But do we want Kube and Load Balancer to be different? Probably not. It would be nice to make UX aware of the discrepancy and get some feedback.
> 
>  Screen.Recording.2024-01-04.at.2.16.14.PM.mov

@mjac0bs 
1. IMO, Opening the drawer is not required in this case, since Regions are not allowed to edit. May be we can update the note that these are defualt regions  where LB will be deployed.
2. I'm inline removing the white background to make it consistent with other places. Also, reduced spacing around the section. (Addressed in [8a7c559](https://github.com/linode/manager/pull/10018/commits/8a7c559409e5621129ad7c732e4f775a49ed051f), [54b41a1](https://github.com/linode/manager/pull/10018/commits/54b41a1a2b41bca7e82e1d741278b707c05dd561))

cpathipa commented:
> Clicking `Edit` on the Configuration causes the Accordion to open/close.
> 
>  Screen.Recording.2024-01-12.at.10.37.30.AM.mov 
> Can we put the headers and the `Edit` buttons on the same line? ![Screenshot 2024-01-12 at 10 46 05 AM](https://private-user-images.githubusercontent.com/115251059/296289863-4204df08-384d-456a-9ad5-340b28b300f6.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwOTEzMDIsIm5iZiI6MTcwNTA5MTAwMiwicGF0aCI6Ii8xMTUyNTEwNTkvMjk2Mjg5ODYzLTQyMDRkZjA4LTM4NGQtNDU2YS05YWQ1LTM0MGIyOGIzMDBmNi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMlQyMDIzMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYzdhNmI2ZjQ3ZmNmZTYxNmY0MDAyOWRkMWQwMjI4MjIyZmJlMzM5OTZhNjgxOGM0YTIzZDMwODhiYjU2MTZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RDrS9kCx_J9-owxJWARAIWdeSUIJY7qFf5Lg0QuOAK8)

@bnussman-akamai 
1. Addressed expanding accordion when clicking on edit button e801f62
2. Addressed Region title alignment [8a7c559](https://github.com/linode/manager/pull/10018/commits/8a7c559409e5621129ad7c732e4f775a49ed051f)

hana-linode commented:
The changes when editing a loadbalancer label or configuration are applying automatically without clicking the Save button

https://github.com/linode/manager/assets/115299789/1a9068cb-a5fa-474a-b9fa-0e6c522be677

Thoughts on removing the first sentence? It seems redundant with the second sentence

![image](https://github.com/linode/manager/assets/115299789/ec4d02b2-abac-4a3e-94da-880ad0b03032)

bnussman-akamai reviewed (commented):
- Can we make the "Regions" section not editable?
  - I don't think users will be able to choose regions for a while longer
  - The Regions drawer has two headers
- In the Figma, the sidebar has a white background (`<Paper/>`) can we make this match what is in Figma?

mjac0bs commented:
> * In the Figma, the sidebar has a white background (`<Paper/>`) can we make this match what is in Figma?

Are we okay with this being inconsistent with Kube Create, where we also have a sidebar, and the background is transparent rather than white?

![image](https://github.com/linode/manager/assets/114685994/854af07c-a992-4a86-a86d-41464b39bb80)

bnussman-akamai commented:
I think so. The transparent background looks a bit 🤮 to me. With the matching white background, I can at least bare having a sidebar in the first place. 


Maybe we can revisit this and make a decision and apply it in all places? @mjac0bs

mjac0bs commented:
> I think so. The transparent background looks a bit 🤮 to me. With the matching white background, I can at least bare having a sidebar in the first place.
> 
> Maybe we can revisit this and make a decision and apply it in all places? @mjac0bs

@bnussman-akamai Yeah, I want us to be consistent with whichever background we do choose. I don't feel very strongly about which one. Thanks for adding to the cafe agenda!

cpathipa commented:
@hana-linode @bnussman-akamai @martinmckenna Great feedback. 

- Removed first sentence in Regions panel (We have UX approval on it.)
- Removed Edit functionality for Regions section.
- For now, I have removed the background color from the summary section. We could revisit this. There is a chance that it might become another panel and may not remain as a sidebar.

hana-linode commented:
@cpathipa The loadbalancer label and routes are still updating without clicking save

mjac0bs reviewed (approved):
Thanks for the updates! Confirmed the changes above, minus the form update issue. I'm approving this so we can get these changes merged pending:

1. We address the typo in the region text. I can't believe I didn't notice it before now! 🔍  `with` should be `will`.

![Screenshot 2024-01-23 at 2 38 02 PM](https://github.com/linode/manager/assets/114685994/3ecd0d58-7432-4073-bf71-9ad4d1918d96)

2. The outstanding issue @hana-linode raised about edit drawers updating before changes are saved is fixed. It would be worth adding some e2e coverage for this too, to check that edits to a label and config is are not saved when the cancel/X button is clicked.

3. If possible, adjusting the alignment of the config Edit button and caret so they're not slightly uneven would be nice, since they're next to each other and it's noticable.

![Screenshot 2024-01-23 at 2 39 17 PM](https://github.com/linode/manager/assets/114685994/e13ffacf-5b4d-461d-b3bf-21eddef3e1ee)

cpathipa commented:
@hana-linode @mjac0bs We can revisit this later. I have update the AC of the following ticket (M3-7297) to address behavior and alignment issues.
